force-app/main/default/classes/StaticTask.cls

```diff
@@ -9,26 +9,32 @@ public with sharing class StaticTask {
   . Then it parses data and return Map collection where key is meail and value is Contact
   Feel free to refactorthis class. Don't forget about single responsibility approach when you create a new method.
   */
+  public static List<String> splitStringBySemicolon(String str) {
+    return str.split(';');
+  }
+  public static String getValue(String str) {
+    return str.split(':')[1].trim();
+  }
+  public static List<String> splitStringByComma(String str) {
+    return str.split(',');
+  }
   public static Map<String, Contact> extractEmailToContactMap(String payload) {
     Map<String, Contact> emailToContactMap = new Map<String, Contact>();
-    List<String> contacts = payload.split(';');
-    for (String contactAsString : contacts) {
+    List<String> contacts = splitStringBySemicolon(payload);
+    for (String contact : contacts) {
```

Comment:
This method has a huge [cyclomatic complexity](https://ru.wikipedia.org/wiki/%D0%A6%D0%B8%D0%BA%D0%BB%D0%BE%D0%BC%D0%B0%D1%82%D0%B8%D1%87%D0%B5%D1%81%D0%BA%D0%B0%D1%8F_%D1%81%D0%BB%D0%BE%D0%B6%D0%BD%D0%BE%D1%81%D1%82%D1%8C).

Please consider splitting it.

I liked the previous implementation more - it was pretty concise on a high level, but had probably too many split. Now it has too less :)

This looked very nice
```
  public static Map<String, Contact> extractEmailToContactMap(String payload) {
    List<String> contacts = getContactsAsStringList(payload);
    setContactsToMap(contacts);

    return emailToContactMap;
  }
````

Reply:
Try starting with extracting each `for` to a separate method. That should give you a good start